### PR TITLE
Add UDP networking support to XRT C++ test

### PR DESCRIPTION
**Pull request by TristanLaan**

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -85,6 +85,6 @@ Open a terminal and run the tests:
 cd "test/host/xrt"
 /bin/cmake .
 make
-mpirun -np <RANKS> bin/test -f -x <XCLBIN FILE>
+mpirun -np <RANKS> bin/test -a -f -x <XCLBIN FILE>
```
Comment (Mellich) [inline INSTALL.md:88]: Guess you also need to add the -a flag to line 76.

**Review by Mellich — CHANGES_REQUESTED** [review]
Tests are working fine, but the test executable is stuck after completion (Like we have observed when using the XRT emulation).

File: INSTALL.md (hunk 1 above)
Comment (TristanLaan) [reply INSTALL.md:88]: No, this is only needed when running on hardware to specify that the xclbin contains 3 CCLOs. In emulation/simulation the xclbin is never loaded, so the flag is simply ignored.

Did emulation not work for you without the -a flag?

Comment (Mellich) [reply INSTALL.md:88]: Sorry, I thought this is always a mandatory option but its only for hardware execution. So everything works as described.

Just the thing with the not exiting the test binary remains.

Comment (TristanLaan) [reply INSTALL.md:88]: Interesting, for me the program does return when running on hardware using both the axis3 and udp run modes. Emulation not exiting is a known bug that's also present in the current dev branch.

Comment (Mellich) [reply INSTALL.md:88]: Weird. Guess its ready to merge then.

**Review by Mellich — APPROVED** [review]

**Pull request merged**